test/functional/tests/memory/test_memory_metadata_consumption.py

```diff
@@ -0,0 +1,191 @@
+        ram_change_status = is_memory_within_limit("Measured increment of RAM usage after start",
+                                                   metadata_max_size, memory_delta)
+
+        TestRun.LOGGER.info("Stop CAS and verify memory changes again")
+        casadm.stop_all_caches()
+
+        free_mem_after_stop = get_mem_available()
+        memory_delta = free_mem_after_stop - free_mem_after_start
+        ram_change_status |= is_memory_within_limit("Measured decrement of RAM usage after stop",
+                                                    metadata_max_size, memory_delta)
+
+        if ram_change_status:
+            TestRun.LOGGER.info(f"Memory measurements completed with correct results.")
+        else:
+            TestRun.LOGGER.error(f"Memory measurements completed, found problems.")
+
+
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_memory_metadata_consumption_full_cache(cache_line_size):
+    """
+        title: Test for CAS RAM consumption for metadata for cache usage 100%.
+        description: |
+          Validate CAS RAM consumption for metadata depends on start cache line size
+          for cache usage 100% (with 5% tolerance) where cache mode Write-Back is used.
+        pass_criteria:
+          - Successful CAS creation and cores addition.
+          - Filling all cache space successfully.
+          - Memory consumption within limits.
+    """
+    with TestRun.step("Prepare drives and partitions for core devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(50, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks['core']
+        parts = [Size(3200, Unit.MebiByte)] * cores_count
+        core_disk.create_partitions(parts)
+        core_devices = core_disk.partitions
+
+    with TestRun.step("Start cache with configuration and add all core devices."):
+        free_mem_before_start = get_mem_available()
+
+        cache = casadm.start_cache(cache_dev, cache_mode_wb, cache_line_size, force=True)
+        Udev.disable()
+        # make sure that I/O will go to the cache
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+        core_list = []
+        for dev in core_devices:
+            core_list.append(cache.add_core(dev))
+
+    with TestRun.step("Fill all cache space - run Fio instance on each of cores"):
+        sync()
+        fio = (Fio().create_command()
+               .direct()
+               .io_engine(IoEngine.libaio)
+               .size(cache.cache_device.size)
+               .read_write(ReadWrite.read)
+               .io_depth(64)
+               .block_size(Size(1, Unit.Blocks4096)))
+        for dev in core_list:
+            fio.add_job(f"job_{dev.core_id}").target(dev.path)
+        fio.run()
```

Comment:
This would read amount of data equal to cache capacity multiplied by number of cores. Instead I would specify size = cache_device.size / num_cores and offset_increment=cache_device.size / num_cores